pypesto/model_selection/model_selection.py

```diff
@@ -0,0 +1,906 @@
+            # TODO remove parameters that are zero
+            pass
+
+        self.model_ID = self.row[MODEL_ID]
+
+        #self.AIC = None
+        self.BIC = None
+
+        if self.valid:
+            self.pypesto_problem = row2problem(row, petab_problem)
+
+            # TODO warning/error if x_fixed_estimated is not a parameter ID in
+            # the PEtab parameter table. A warning is currently produced in
+            # `row2problem` above.
+            # Could move to a separate method that is only called when a
+            # criterion that requires the number of estimated parameters is
+            # called.
+            self.estimated = x_fixed_estimated | set(
+                self.petab_problem.parameter_df.query(f'{ESTIMATE} == 1').index
+            )
+            self.n_estimated = len(self.estimated)
+            # TODO for BIC. Could move to `calculate_BIC()` to save time.
+            self.n_edata = None
+
+            self.minimize_result = None
+
+            if autorun:
+                self.set_result(minimize(self.pypesto_problem))
+
+    def set_result(self, result: Result):
+        self.minimize_result = result
+        self.optimized_model = self.minimize_result.optimize_result.list[0]
+
+    @property
+    def AIC(self):
+        if '_AIC' not in dir(self):
+            self._AIC = calculate_AIC(self.n_estimated,
+                                      self.optimized_model.fval)
+        return self._AIC
+
+    def calculate_BIC(self):
+        # TODO: test, implement self.n_data in `__init__`
+        # TODO: find out how to get size of experimental data
+        #       petab.problem
+        raise NotImplementedError
+        return self.n_estimated*math.log(self.n_data) + \
+            2*self.minimize_result[0].fval
+            #-2*math.log(self.minimize_result[0].fval)
+
+    #def calibrate(self):
+    #    # TODO settings?
+    #    # 100 starts, SingleCoreEngine, ScipyOptimizer
+    #    result = minimize(self.pypesto_problem)
+    #    self.set_result(result)
+
+
+class ModelSelector:
+    def __init__(
+            self,
+            petab_problem: petab.problem,
+            specification_filename: str,
+            #constraints_filename: str
+    ):
+        self.petab_problem = petab_problem
+        # TODO remove duplicates from specification_file
+        self.specification_file = unpack_file(specification_filename)
+        self.header = line2row(self.specification_file.readline(),
+                               convert_parameters_to_float=False)
+        self.parameter_IDs = self.header[PARAMETER_DEFINITIONS_START:]
+
+
+        #self.apply_constraints(
+        #    self.parse_constraints_file[constraints_filename])
+
+        self.selection_history = {}
+
+        # Dictionary of method names as keys, with a dictionary as the values.
+        # In the dictionary, keys will be modelId, criterion value
+        # TODO unused
+        self.results = {}
+
+    def delete_models(self,
```

Comment:
if the constraints are to be checked in here, better call it `delete_if_constraints_fail` or sth like that.